### PR TITLE
maven_artifact: download with shutil.copyfileobj to a temp file

### PR DESCRIPTION
**Pull request by turb**
##### SUMMARY
Download in the same fashion than `get_url`:
- Replace the homemade chunked download by the more standard `shutil.copyfileobj`
- Copy to a temp file to prevent having a corrupted file in case of download error

Should replace #42844.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
maven_artifact

##### ANSIBLE VERSION

```
ansible 2.6.3
  config file = None
  configured module search path = [u'/Users/turb/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.6.3/libexec/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### ADDITIONAL INFORMATION

Tested with Artifactory, not with other repositories (and not with local repo).

### CODE REVIEW

File: lib/ansible/modules/packaging/language/maven_artifact.py [hunk 1]
```diff
@@ -371,56 +371,41 @@ def _request(self, url, failmsg, force=True):
...
+                if invalid_md5:
+                    # if verify_change was set, the previous file would be deleted
+                    os.remove(tempname)
+                    return invalid_md5
+        except Exception as e:
+            os.remove(tempname)
+            raise e
+
+        # all good, now copy temp file to target
+        shutil.copy2(tempname, artifact.get_filename(filename))
```
Comment (tumbl3w33d) [inline lib/ansible/modules/packaging/language/maven_artifact.py]: How about `shutil.move` here?

Comment (turb) [reply lib/ansible/modules/packaging/language/maven_artifact.py]: You're damn right.

**Comment by turb** [comment]
@tumbl3w33d amended

**Comment by tumbl3w33d** [comment]
Did a quick check with Nexus 3, works.

shipit

**Pull request merged**